components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
```

Comment:
It looks like you may be confusing State with InputStates.  AFAICT this module should only deal with a single State object, which has properties: data, lm, previousLms, oldLm, port.  See slide 9: http://dbooth.org/2013/pipeline/RDF-Pipeline-Implementation.pdf